insights/parsers/cryptsetup_luksDump.py

```diff
@@ -0,0 +1,171 @@
+===========================================
+This class provides parsing for the output of cryptsetup luksDump
+<device_name>. Outputs from LUKS1 and LUKS2 are supported.
+"""
+
+from insights import parser, Parser, SkipComponent
+from insights.parsers import ParseException
+from insights.parsr import (Literal, AnyChar, Char, Opt, String, WS,
+                            HangingString, WithIndent, Many, EOF, Lift)
+from insights.specs import Specs
+
+import string
+
+
+class DocParser(object):
+    def __init__(self):
+        value_chars = set(string.printable) - set("\n\r")
+
+        FirstLine = Literal("LUKS header information", value="header") << AnyChar.until(Char("\n")) + Opt(Many(Char("\n")))
+        FirstIndent = Literal("  ")
+        # we need to replace the \t by 8 spaces in the input,
+        # otherwise WithIndent does not work properly
+        # SecondIndent = Literal("\t")
+        SecondIndent = Literal(" " * 8)
+
+        Key = String(value_chars - set(":")) << Char(":") % "Key"
+        Value = WS >> HangingString(value_chars) % "Value"
+
+        MultilineContinuation = Many((Char("\n") + Literal(" " * 15) + SecondIndent) >> String(value_chars))
+        Value1 = WS >> String(value_chars) + Opt(MultilineContinuation).map(lambda x: "".join(x)) << Char("\n")
+        Value1 = Value1.map(lambda x: ("".join(x)).strip())
+
+        ZeroLevelKVPair = Key + Value1
+        FirstLevelKVPair = FirstIndent >> Key + Value1
+        SecondLevelKVPair = SecondIndent >> WithIndent(Key + Value)
+
+        Luks2SectionName = Key << Char("\n")
+        Luks2SectionEntry = (FirstLevelKVPair + Many(SecondLevelKVPair).map(dict)).map(self.convert_type)
+        Luks2Section = Luks2SectionName + Many(Luks2SectionEntry).map(dict) << Opt(Many(Char("\n")))
+        Luks2Body = Many(Luks2Section, lower=1)
+
+        Luks1Section = ZeroLevelKVPair + Many(SecondLevelKVPair).map(dict) << Opt(Many(Char("\n")))
+        Luks1Body = Many(Luks1Section.map(self.convert_status), lower=1)
+
+        KVBlock = Many(Key + Value1).map(dict)
+        Luks_Header = (FirstLine + KVBlock) << Opt(Many(Char("\n")))
+
+        # Luks2Body has to go first, because Luks1Body consumes also valid Luks2 bodies
+        self.Top = Lift(self.join_header_and_body) * Luks_Header * (Luks2Body | Luks1Body) << EOF
+
+    def join_header_and_body(self, header, body):
+        return dict([header] + body)
+
+    def convert_type(self, section):
+        section[1]["type"] = section[0][1]
+        return [section[0][0], section[1]]
+
+    def convert_status(self, section):
+        section[2]["status"] = section[1]
+        return [section[0], section[2]]
+
+    def __call__(self, content):
+        try:
+            return self.Top(content)
+        except Exception:
+            raise ParseException("There was an exception when parsing one of the outputs of cryptsetup luksDump commands.")
+
+
+@parser(Specs.cryptsetup_luksDump)
+class Luks_dump(Parser):
```

Comment:
@danzatt - Thanks for the update, but I think you may misunderstand the `CapsWords`. Would you please change the class to `LuksDump`?

Reply:
Sorry, done now.